synapse/lib/stormtypes.py

```diff
@@ -3654,50 +3661,69 @@ def getObjLocals(self):
             'gen': self._methQueueGen,
             'del': self._methQueueDel,
             'get': self._methQueueGet,
+            'byname': self._methQueueGetByName,
             'list': self._methQueueList,
         }
 
-    async def _methQueueAdd(self, name):
+    async def _methQueueAdd(self, name, iden=None):
+        name = await tostr(name)
+        iden = await tostr(iden, noneok=True)
 
-        info = {
-            'time': s_common.now(),
+        qdef = {
             'creator': self.runt.user.iden,
+            'iden': iden,
```

Comment:
Remove this line and do the assignment in the `if iden is not None` block below.